### PR TITLE
Extended errors

### PR DESCRIPTION
**Pull request by yndc**
This is a PR for adding feature described in #144 

I'd like for some input on my implementation and naming. 
- Is `FastifyGraphQLError` class for the extended error suitable?
- How should we structure the error parameters? At the moment I'm using Apollo's `ApolloError` format; i.e. `(message: string, code: string, additionalProperties: object)`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -361,6 +363,15 @@ module.exports = fp(async function (app, opts) {
     )
 
     if (execution.errors) {
+      execution.errors = execution.errors.map(e => {
+        if (e.originalError instanceof FastifyGraphQLError) {
```
Comment (mcollina) [inline index.js:367]: I would prefer if we duck-typed this instead of checking for inheritance, so any error with `code` and `additionalProperties`   would work.

File: lib/routes.js [hunk 2]
```diff
@@ -37,10 +37,7 @@ const responseSchema = {
             },
             extensions: {
               type: 'object',
-              properties: {
-                code: { type: 'string' },
-                timestamp: { type: 'string' }
-              }
+              additionalProperties: true
```
Comment (yndc) [inline lib/routes.js:40]: Would this cause security issues?

File: lib/errors.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  constructor (message, code, additionalProperties) {
+    super(message)
+    this.code = code
+    this.additionalProperties = additionalProperties
+  }
+}
+
+module.exports = {
+  FastifyGraphQLError: FastifyGraphQLError
+}
```
Comment (mcollina) [inline lib/errors.js:13]: I would call this `ErrorWithProps` or similar and not have the name of the framework in it.

File: lib/routes.js [hunk 4]
```diff
@@ -37,10 +37,7 @@ const responseSchema = {
             },
             extensions: {
               type: 'object',
-              properties: {
-                code: { type: 'string' },
-                timestamp: { type: 'string' }
-              }
```
Comment (mcollina) [inline lib/routes.js:43]: why were this removed?

Comment (yndc) [reply lib/routes.js:43]: Hmm, yes I'll re-add `code`, but should `timestamp` stays? I think it shouldn't be hard-coded as a string. There might be an use-case where a developer want's `extensions.timestamp` to be a `number` type.

**Comment by yndc** [comment]
Should we use the current error constructor (with `code`) or just let the developer shape everything inside `extensions` themselves to make it less opinionated?

Then `ErrorWithProps` constructor will only be `new ErrorWithProps(message: string, additionalProperties: object)`

File: lib/routes.js (hunk 2 above)
Comment (mcollina) [reply lib/routes.js:41]: no this is ok.

File: lib/routes.js [hunk 5]
```diff
@@ -37,10 +37,8 @@ const responseSchema = {
...
+              code: { type: 'string' },
```
Comment (mcollina) [inline lib/routes.js:40]: this needs to be in properties

Comment (yndc) [reply lib/routes.js]: whoops! missed that
